tools/sklearn/main_macros.xml

```diff
@@ -104,11 +99,70 @@ def get_X_y(params, file1, file2):
   return X, y
   </token>
 
+  <token name="@GET_SEARCH_PARAMS_FUNCTION@">
+def get_search_params(params_builder):
+
+  search_params = {}
+
+  def safe_eval(literal):
+    scipy_stats = getattr(scipy, 'stats')
+    syms = make_symbol_table(use_numpy=True, scipy_stats=scipy_stats)
+    if literal.startswith("scipy.stats"):
+      literal = literal.replace("scipy.stats", "scipy_stats")
+    return Interpreter(syms)(literal)
+
+  def add_pipeline_params(search_params, param_set, param_name, param_pre):
+    for p in params_builder[param_set]:
+      if p[param_name].strip() == '':
+        continue
+      lst = p[param_name].split(":")
+      assert (len(lst) == 2), "Error, make sure there is one and only one colon in search parameter input."
+      literal = lst[1].strip().replace("__sq__","'")\
+          .replace('__dq__','"').replace("__oc__", "{").replace("__cc__", "}")\
+          .replace("__ob__", "[").replace("__cb__", "]")
+      try:
+        ev = safe_eval(literal)
+      except:
+        sys.exit("Unsupported parameter input: %s" %p[param_name])
+      search_params[param_pre + lst[0].strip()] = ev
+
+  add_pipeline_params(search_params, 'estimator_param_set', 'estimator_param', 'estimator__')
+  add_pipeline_params(search_params, 'prep_param_set', 'prep_param', 'pre_processor__')
+  add_pipeline_params(search_params, 'fs_param_set', 'fs_param', 'feature_selector__')
+  return search_params
+  </token>
+
+  <token name="@GET_ESTIMATOR_FUNCTION@">
+def get_estimator(estimator_json):
+  estimator_module = estimator_json['selected_module']
+  estimator_cls = estimator_json['selected_estimator']
+
+  if estimator_module == "xgboost":
+    cls = getattr(xgboost, estimator_cls)
+  else:
+    module = getattr(sklearn, estimator_module)
+    cls = getattr(module, estimator_cls)
+
+  estimator_params = estimator_json['estimator_params'].strip()
+  if estimator_params == "":
+    estimator = cls()
+  else:
+    estimator_params = estimator_params.replace("__sq__","'")
+    try:
+      params = ast.literal_eval('{' + estimator_params + '}')
```

Comment:
this we can also replace I think?

Reply:
Do you have a reference for it?

Reply:
Oh sorry, I mixed it up ... narf. Too late.

Reply:
`literal_eval` should be a safe bet, as safe as python.
https://stackoverflow.com/questions/4710247/python-3-are-there-any-known-security-holes-in-ast-literal-evalnode-or-string